app/controllers/urls_controller.rb

```diff
@@ -0,0 +1,27 @@
+class UrlsController < ApplicationController
+
+  def index #('/')
+    @url = Url.new
+    @urls = Url.all.order(:created_at)
+  end
+
+  def create #(post '/urls')
+    @new_url = Url.new(long: params[:url][:long])
+
+      if @new_url.save
+        redirect_to '/'
+      else
+        [400, "Url not Converted"]
```

Comment:
So, I know we were like NEVER EVER EVER RENDER IN A POST. But actually, that was a lie. What you should never do is render in a successful post response. It's fine to render on an error (because the browser won't cache it).

So you can replace this `else` block with `render(:index, status: :bad_request)`